library/ssl_tls.c

```diff
@@ -5428,7 +5439,7 @@ static int ssl_context_load(mbedtls_ssl_context *ssl,
             /* alpn_chosen should point to an item in the configured list */
             for (cur = ssl->conf->alpn_list; *cur != NULL; cur++) {
                 if (strlen(*cur) == alpn_len &&
-                    memcmp(p, cur, alpn_len) == 0) {
+                    memcmp(p, *cur, alpn_len) == 0) {
```

Comment:
Good to fix that but please do it in a separate commit and add a bugfix change log. We probably need a backport for that.